compiler/z/codegen/UnaryEvaluator.cpp

```diff
@@ -520,9 +520,14 @@ OMR::Z::TreeEvaluator::l2aEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    node->setRegister(source);
 
 
-   //printf("Node %p is first child\n", firstChild); fflush(stdout);
-   //traceMsg(comp, "Node %p is first child\n", firstChild);
-   cg->decReferenceCount(firstChild);
+   if (shouldGenerateGuardedLoadsForCompressedLoad)
+      {
+      cg->decReferenceCount(firstChild->getFirstChild());
```

Comment:
When we generate guarded load we don't need to do a shift, the result from load is shifted. This is a way to skip it. Is there a better way to skip evaluating the shift, do I need to decrement fistChild (shift node) if we never evaluate it?